src/services/validateZendeskRequest.spec.ts

```diff
@@ -0,0 +1,412 @@
+  it('should parse data into response if request contains sessionIds', () => {
+    const validationResult = validateZendeskRequest(
+      JSON.stringify(
+        buildValidRequestBodyWithIds(
+          'session_id',
+          'sessionId1 sessionId2 sessionId3'
+        )
+      )
+    )
+
+    expect(validationResult.dataRequestParams?.sessionIds).toEqual([
+      'sessionId1',
+      'sessionId2',
+      'sessionId3'
+    ])
+    expect(validationResult.dataRequestParams?.journeyIds).toBeUndefined()
+    expect(validationResult.dataRequestParams?.eventIds).toBeUndefined()
+  })
+
+  it('should return an invalid response if request does not contain sessionIds when required', () => {
+    const requestBody = buildValidRequestBody()
+    requestBody.identifierType = 'session_id'
+    requestBody.sessionIds = ''
+    const validationResult = validateZendeskRequest(JSON.stringify(requestBody))
+    expect(validationResult.isValid).toEqual(false)
+    expect(validationResult.validationMessage).toEqual(
+      'At least one session id should be provided'
+    )
+  })
+
+  it('should return an invalid response if request does not contain journeyIds when required', () => {
+    const requestBody = buildValidRequestBody()
+    requestBody.identifierType = 'journey_id'
+    requestBody.journeyIds = ''
+    const validationResult = validateZendeskRequest(JSON.stringify(requestBody))
+    expect(validationResult.isValid).toEqual(false)
+    expect(validationResult.validationMessage).toEqual(
+      'At least one journey id should be provided'
+    )
+  })
+
+  it('should return an invalid response if request does not contain eventIds when required', () => {
+    const requestBody = buildValidRequestBody()
+    requestBody.identifierType = 'event_id'
+    requestBody.eventIds = ''
+    const validationResult = validateZendeskRequest(JSON.stringify(requestBody))
+    expect(validationResult.isValid).toEqual(false)
+    expect(validationResult.validationMessage).toEqual(
+      'At least one event id should be provided'
+    )
+  })
+
+  it('should parse data into response if request contains journeyIds', () => {
+    const validationResult = validateZendeskRequest(
+      JSON.stringify(
+        buildValidRequestBodyWithIds(
+          'journey_id',
+          'journeyId1 journeyId2 journeyId3'
+        )
+      )
+    )
+
+    expect(validationResult.dataRequestParams?.journeyIds).toEqual([
+      'journeyId1',
+      'journeyId2',
+      'journeyId3'
+    ])
+    expect(validationResult.dataRequestParams?.eventIds).toBeUndefined()
+    expect(validationResult.dataRequestParams?.sessionIds).toBeUndefined()
+  })
+
+  it('should parse data into response if request contains eventIds', () => {
+    const validationResult = validateZendeskRequest(
+      JSON.stringify(buildValidRequestBodyWithIds('event_id', 'id1 id2 id3'))
+    )
+
+    expect(validationResult.dataRequestParams?.eventIds).toEqual([
+      'id1',
+      'id2',
+      'id3'
+    ])
+  })
+
```

Comment:
There is the `User ID` identifier also on the Zendesk form. Should we be checking for that as well?

Reply:
This has been added and ready for review.

Reply:
> There is the `User ID` identifier also on the Zendesk form. Should we be checking for that as well?

This has been added